### PR TITLE
Generate Azure Eventgrid SDK for 2023-06-01-preview

### PR DESCRIPTION
**Pull request by rohkuma-microsoft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/src/Generated/SubscriptionResource.cs [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+namespace Azure.ResourceManager.EventGrid
+{
+    /// <summary>
+    /// A Class representing a Subscription along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="SubscriptionResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetSubscriptionResource method.
+    /// Otherwise you can get one from its parent resource <see cref="NamespaceTopicResource" /> using the GetSubscription method.
+    /// </summary>
+    public partial class SubscriptionResource : ArmResource
```
Comment (live1206) [inline sdk/eventgrid/Azure.ResourceManager.EventGrid/src/Generated/SubscriptionResource.cs:26]: We need to rename `SubscriptionResource` to a better name, it is conflicting with [Azure.ResourceManager.Resources.SubscriptionResource](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/src/Resources/Generated/SubscriptionResource.cs), and also [EventGridSubscriptionData](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Azure.ResourceManager.EventGrid/src/Generated/EventGridSubscriptionData.cs) is in use.

Add below config under rename-mapping in autorest.md
```
Subscription: ABetterNameBasedOnTheUsage
```

Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/src/Generated/SubscriptionResource.cs:26]: fixed

Comment (live1206) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/src/Generated/SubscriptionResource.cs:26]: This file still got generated, not resolved.
And it is causing the conflicts with [Azure.ResourceManager.Resources.SubscriptionResource](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/src/Resources/Generated/SubscriptionResource.cs), also generating wrong operation methods in `EventGridExtensions.cs`, ending up to build errors.

Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/src/Generated/SubscriptionResource.cs]: fixed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,21 @@
 
 ### Other Changes
 
+## 1.1.0-beta.2 (2023-06-30)
+
+### Features Added
+- Add support for Namespace
+- Add support for Namespace Topics.
+- Add support for Namespace Topic subscriptions.
```
Comment (ahamad-MS) [inline sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md:18]: - Add support for Namespace Topic EventSubscriptions.

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,21 @@
...
+- Add support for pull events from Namespace topics.
+- Add support for MQTT scenarios.
```
Comment (ahamad-MS) [inline sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md:20]: Explicitly determine the nested namespace resources that are mqtt related.. e.g., clients, caCertificates, etc.

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md (hunk 2 above)
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md]: fixed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md (hunk 3 above)
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md]: fixed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.1.0-beta.2 (Unreleased)
+## 1.1.0-beta.3 (Unreleased)
```
Comment (live1206) [inline sdk/eventgrid/Azure.ResourceManager.EventGrid/CHANGELOG.md:3]: Please remove the section of `1.1.0-beta.3`, the latest version should be `1.1.0-beta.2`, that's why CI failed.

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs [hunk 5]
```diff
@@ -1151,10 +1794,13 @@ public partial class EventGridDomainPatch
         public bool? AutoCreateTopicWithFirstSubscription { get { throw null; } set { } }
         public bool? AutoDeleteTopicWithLastSubscription { get { throw null; } set { } }
         public Azure.ResourceManager.EventGrid.Models.DataResidencyBoundary? DataResidencyBoundary { get { throw null; } set { } }
+        public Azure.ResourceManager.EventGrid.Models.PartnerTopicEventTypeInfo EventTypeInfo { get { throw null; } set { } }
         public Azure.ResourceManager.Models.ManagedServiceIdentity Identity { get { throw null; } set { } }
         public System.Collections.Generic.IList<Azure.ResourceManager.EventGrid.Models.EventGridInboundIPRule> InboundIPRules { get { throw null; } }
         public bool? IsLocalAuthDisabled { get { throw null; } set { } }
+        public Azure.ResourceManager.EventGrid.Models.TlsVersion? MinimumTlsVersionAllowed { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs:1754]: This is not a boolean check. It takes the actual value of TLS version like 1.1, 1.2, etc

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs [hunk 6]
```diff
@@ -1043,6 +1500,146 @@ public partial class BoolEqualsAdvancedFilter : Azure.ResourceManager.EventGrid.
...
+        public override bool Equals(object obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(Azure.ResourceManager.EventGrid.Models.ClientProvisioningState left, Azure.ResourceManager.EventGrid.Models.ClientProvisioningState right) { throw null; }
+        public static implicit operator Azure.ResourceManager.EventGrid.Models.ClientProvisioningState (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.EventGrid.Models.ClientProvisioningState left, Azure.ResourceManager.EventGrid.Models.ClientProvisioningState right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ClientState : System.IEquatable<Azure.ResourceManager.EventGrid.Models.ClientState>
```
[the remark this answers is not in the document]
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs]: Renamed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs [hunk 7]
```diff
@@ -1043,6 +1500,146 @@ public partial class BoolEqualsAdvancedFilter : Azure.ResourceManager.EventGrid.
...
+        public override bool Equals(object obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(Azure.ResourceManager.EventGrid.Models.ClientGroupProvisioningState left, Azure.ResourceManager.EventGrid.Models.ClientGroupProvisioningState right) { throw null; }
+        public static implicit operator Azure.ResourceManager.EventGrid.Models.ClientGroupProvisioningState (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.EventGrid.Models.ClientGroupProvisioningState left, Azure.ResourceManager.EventGrid.Models.ClientGroupProvisioningState right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ClientProvisioningState : System.IEquatable<Azure.ResourceManager.EventGrid.Models.ClientProvisioningState>
```
[the remark this answers is not in the document]
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs]: Renamed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs [hunk 8]
```diff
@@ -531,6 +672,158 @@ public partial class ExtensionTopicResource : Azure.ResourceManager.ArmResource
         public virtual Azure.Response<Azure.ResourceManager.EventGrid.ExtensionTopicResource> Get(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.EventGrid.ExtensionTopicResource>> GetAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    public partial class NamespaceCollection : Azure.ResourceManager.ArmCollection, System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.EventGrid.NamespaceResource>, System.Collections.Generic.IEnumerable<Azure.ResourceManager.EventGrid.NamespaceResource>, System.Collections.IEnumerable
```
[the remark this answers is not in the document]
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs]: Renamed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs [hunk 9]
```diff
@@ -1,5 +1,119 @@
...
+        public ClientData() { }
+        public System.Collections.Generic.IDictionary<string, System.BinaryData> Attributes { get { throw null; } }
+        public Azure.ResourceManager.EventGrid.Models.ClientAuthentication Authentication { get { throw null; } set { } }
+        public string AuthenticationName { get { throw null; } set { } }
+        public Azure.ResourceManager.EventGrid.Models.ClientCertificateAuthentication ClientCertificateAuthentication { get { throw null; } set { } }
+        public string Description { get { throw null; } set { } }
+        public Azure.ResourceManager.EventGrid.Models.ClientProvisioningState? ProvisioningState { get { throw null; } }
+        public Azure.ResourceManager.EventGrid.Models.ClientState? State { get { throw null; } set { } }
+    }
+    public partial class ClientGroupCollection : Azure.ResourceManager.ArmCollection, System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.EventGrid.ClientGroupResource>, System.Collections.Generic.IEnumerable<Azure.ResourceManager.EventGrid.ClientGroupResource>, System.Collections.IEnumerable
```
[the remark this answers is not in the document]
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs]: Renamed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs [hunk 10]
```diff
@@ -1,5 +1,119 @@
...
+        public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.EventGrid.CaCertificateResource> Update(Azure.WaitUntil waitUntil, Azure.ResourceManager.EventGrid.CaCertificateData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.EventGrid.CaCertificateResource>> UpdateAsync(Azure.WaitUntil waitUntil, Azure.ResourceManager.EventGrid.CaCertificateData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+    }
+    public partial class ClientCollection : Azure.ResourceManager.ArmCollection, System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.EventGrid.ClientResource>, System.Collections.Generic.IEnumerable<Azure.ResourceManager.EventGrid.ClientResource>, System.Collections.IEnumerable
```
[the remark this answers is not in the document]
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs]: Renamed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs [hunk 11]
```diff
@@ -1043,6 +1500,146 @@ public partial class BoolEqualsAdvancedFilter : Azure.ResourceManager.EventGrid.
...
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(Azure.ResourceManager.EventGrid.Models.CaCertificateProvisioningState left, Azure.ResourceManager.EventGrid.Models.CaCertificateProvisioningState right) { throw null; }
+        public static implicit operator Azure.ResourceManager.EventGrid.Models.CaCertificateProvisioningState (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.EventGrid.Models.CaCertificateProvisioningState left, Azure.ResourceManager.EventGrid.Models.CaCertificateProvisioningState right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class ClientAuthentication
```
[the remark this answers is not in the document]
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs]: renamed

File: sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs [hunk 12]
```diff
@@ -816,6 +1159,44 @@ public partial class PartnerTopicResource : Azure.ResourceManager.ArmResource
         public virtual Azure.Response<Azure.ResourceManager.EventGrid.PartnerTopicResource> Update(Azure.ResourceManager.EventGrid.Models.PartnerTopicPatch patch, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.EventGrid.PartnerTopicResource>> UpdateAsync(Azure.ResourceManager.EventGrid.Models.PartnerTopicPatch patch, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    public partial class PermissionBindingCollection : Azure.ResourceManager.ArmCollection, System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.EventGrid.PermissionBindingResource>, System.Collections.Generic.IEnumerable<Azure.ResourceManager.EventGrid.PermissionBindingResource>, System.Collections.IEnumerable
```
[the remark this answers is not in the document]
Comment (rohkuma-microsoft) [reply sdk/eventgrid/Azure.ResourceManager.EventGrid/api/Azure.ResourceManager.EventGrid.netstandard2.0.cs]: renamed

**Review by live1206 — APPROVED** [review]

**Pull request merged**